actionview/lib/action_view/template/resolver.rb

```diff
@@ -99,6 +90,8 @@ def templates_have_changed?(cached_templates, fresh_templates)
         # compare modification times, and instead just check whether the lists are different
         if cached_templates.blank? || fresh_templates.blank?
           return fresh_templates.blank? != cached_templates.blank?
+        elsif cached_templates.size != fresh_templates.size
+          return true
```

Comment:
Found issues when adding templates in tests. Because the new template was added within a second of the existing files, the timestamp matching below would bork.

Not sure how well checking for size meshes with real life (and if we should just add sleeps in the tests instead).